sw/device/demos/aes_hmac/aes_hmac_demo.c

```diff
@@ -0,0 +1,166 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/aes.h"
+#include "sw/device/lib/common.h"
+#include "sw/device/lib/uart.h"
+
+#define LENGTH_KEY 32
+#define MAX_LENGTH_TEXT 128
+
+/**
+ * Wait for AES unit to have valid output data and then
+ * get one 16B block of output data from AES unit.
+ *
+ * @param input_string pointer to string buffer.
+ * @param length_max   size of the string buffer.
+ */
+void overwrite_string(char *input_string, size_t length_max) {
+  bool done = false;
+  int ret = -1;
+  char my_char = 0;
+  int index = 0;
+
+  while (done == false) {
+    ret = uart_rcv_char(&my_char);
+
+    if (ret == 0) {
+      if (my_char == 13) {
```

Comment:
maybe a comment that `13` (or `0x0d`) is the carriage return

Reply:
Done.